aws-predictions/src/main/java/com/amplifyframework/predictions/aws/service/AWSPollyService.java

```diff
@@ -0,0 +1,116 @@
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.predictions.aws.service;
+
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.core.Consumer;
+import com.amplifyframework.predictions.PredictionsException;
+import com.amplifyframework.predictions.aws.AWSPredictionsPluginConfiguration;
+import com.amplifyframework.predictions.aws.configuration.SpeechGeneratorConfiguration;
+import com.amplifyframework.predictions.aws.models.VoiceType;
+import com.amplifyframework.predictions.result.TextToSpeechResult;
+import com.amplifyframework.util.UserAgent;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.mobile.client.AWSMobileClient;
+import com.amazonaws.services.polly.AmazonPollyClient;
+import com.amazonaws.services.polly.AmazonPollyPresigningClient;
+import com.amazonaws.services.polly.model.OutputFormat;
+import com.amazonaws.services.polly.model.SynthesizeSpeechRequest;
+import com.amazonaws.services.polly.model.SynthesizeSpeechResult;
+import com.amazonaws.services.polly.model.TextType;
+
+import java.io.InputStream;
+
+/**
+ * Predictions service for performing text to speech conversion.
+ */
+final class AWSPollyService {
+    private static final int MP3_SAMPLE_RATE = 24_000;
+
+    private final AmazonPollyClient polly;
+    private final AWSPredictionsPluginConfiguration pluginConfiguration;
+
+    AWSPollyService(AWSPredictionsPluginConfiguration pluginConfiguration) {
+        this.polly = createPollyClient();
+        this.pluginConfiguration = pluginConfiguration;
+    }
+
+    private AmazonPollyClient createPollyClient() {
+        AWSCredentialsProvider credentialsProvider = AWSMobileClient.getInstance();
+        ClientConfiguration configuration = new ClientConfiguration();
+        configuration.setUserAgent(UserAgent.string());
+        return new AmazonPollyPresigningClient(credentialsProvider, configuration);
+    }
+
+    void synthesizeSpeech(
+            @NonNull String text,
+            @NonNull VoiceType voiceType,
+            @NonNull Consumer<TextToSpeechResult> onSuccess,
+            @NonNull Consumer<PredictionsException> onError
+    ) {
+        try {
+            InputStream data = synthesizeSpeech(text, voiceType);
+            onSuccess.accept(TextToSpeechResult.fromAudioData(data));
+        } catch (PredictionsException exception) {
+            onError.accept(exception);
+        }
+    }
+
+    private InputStream synthesizeSpeech(String text, VoiceType voiceType) throws PredictionsException {
+        final String languageCode;
+        final String voiceId;
+        if (VoiceType.UNKNOWN.equals(voiceType)) {
```

Comment:
Could you represent an unknown VoiceType with `null` instead of a `UNKNOWN` type?

Reply:
I'd prefer to leave all instances where a `VoiceType` is expected to be `@NonNull` as to minimize null-checking